restapi/src/main/java/io/stargate/web/docsapi/service/query/FieldsFilter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.web.docsapi.service.query;
+
+import io.stargate.db.datastore.Row;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.service.DocsApiConfiguration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class FieldsFilter {
+  private final DocsApiConfiguration config;
+
+  public FieldsFilter(DocsApiConfiguration config) {
+    this.config = config;
+  }
+
+  public List<Row> filter(List<Row> rows, List<String> fieldNames, List<String> requestedPath) {
```

Comment:
this is very very similar to:

* https://github.com/stargate/stargate/pull/1047/files#diff-7e2c32d1a1e0d69029375dca50a0fc34d4ff13d8cf37adc1bc44d8521aac6db9R188
* and https://github.com/stargate/stargate/pull/1047/files#diff-cdeb981391b91ebd4380273c4796ef53a64c1e810588e5aaab3480d2a7bdf200L128

Let's somehow unify this..

Reply:
Yeah... I was working on an earlier version of the code and rebased only at the last min :facepalm:

Reply:
+1 to unifying, but I'm not sure the functionality is exactly the same.

For example, this filter will reject nested fields, but the linked `isRowOnPath` allows nested values. WDYT?

Reply:
Rebased to reuse `DocApiUtils`. The `FieldsFilter` class got removed during rebasing.